pydantic_ai_slim/pydantic_ai/providers/google_vertex.py

```diff
@@ -97,15 +97,21 @@ def __init__(
         if service_account_file and service_account_info:
             raise ValueError('Only one of `service_account_file` or `service_account_info` can be provided.')
 
-        self._client = http_client or cached_async_http_client(provider='google-vertex')
         self.service_account_file = service_account_file
         self.service_account_info = service_account_info
         self.project_id = project_id
         self.region = region
         self.model_publisher = model_publisher
 
+        if http_client is not None:
+            self._client = http_client
+            if str(http_client.base_url) == '':
+                self._client.base_url = self.base_url
+        else:
+            self._client = cached_async_http_client(provider='google-vertex')
+            self._client.base_url = self.base_url
```

Comment:
I don't think this makes sense for Vertex. The auth will really only work for Google.